src/Symfony/Component/Filesystem/Tests/FilesystemTest.php

```diff
@@ -375,9 +375,18 @@ public function testFilesExists()
         mkdir($basePath);
         touch($basePath.'file1');
         mkdir($basePath.'folder');
+        $this->assertTrue($this->filesystem->exists($basePath.'file1', false));
+        $this->assertTrue($this->filesystem->exists($basePath.'folder', false));
+    }
 
-        $this->assertTrue($this->filesystem->exists($basePath.'file1'));
-        $this->assertTrue($this->filesystem->exists($basePath.'folder'));
+    public function testDanglingExists()
+    {
+        $basePath = $this->workspace.\DIRECTORY_SEPARATOR.'directory'.\DIRECTORY_SEPARATOR;
+
+        mkdir($basePath);
+        touch($basePath.'testLink');
+        exec(sprintf('ln -s %stestLink %stestDanglingLink && rm %stestLink', $basePath, $basePath, $basePath));
+        $this->assertTrue($this->filesystem->exists($basePath.'testDanglingLink', true));
```

Comment:
we should also test `exists($path)`  on a dangling symlink IMHO

Reply:
done